src/evict/evict_page.c

```diff
@@ -675,9 +676,40 @@ __evict_review(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t evict_flags, bool
             LF_SET(WT_REC_SCRUB);
     }
 
+    /* Acquire a snapshot if coming through eviction thread route. */
+    if (F_ISSET(session, WT_SESSION_INTERNAL) && !F_ISSET(session->txn, WT_TXN_RUNNING) &&
+      !F_ISSET(conn,
+        WT_CONN_CLOSING | WT_CONN_RECOVERING | WT_CONN_IN_MEMORY | WT_CONN_CLOSING_TIMESTAMP) &&
+      !WT_IS_HS(S2BT(session))) {
+        /*
+         * We don't have a running transaction with eviction threads. Starting a new transaction
+         * should implicitly take a snapshot as well.
+         */
+        WT_RET(__wt_txn_begin(session, txn_cfg));
```

Comment:
Why are we allocating a full transaction here, not just allocating a snapshot?

Reply:
I started off with only having a snapshot rather than a full fledge transaction. There were few asserts that fired up complaining about doing visibility checks without having an active transaction.